src/Corax/IndexSearcher.cs

```diff
@@ -581,6 +636,189 @@ public MultiTermMatch ContainsQuery<TScoreFunction>(string field, string contain
+                    mode |= SearchMatchOptions.EndsWith;
+                if (searchTerm[^1] is '*')
+                    mode |= SearchMatchOptions.StartsWith;
+                if (mode.HasFlag(SearchMatchOptions.StartsWith | SearchMatchOptions.EndsWith))
+                    mode = SearchMatchOptions.Contains;
+
+                var encodedString = Encodings.Utf8.GetString(word);
+                switch (mode)
+                {
+                    case SearchMatchOptions.Exact:
+                        if (match is null)
+                        {
+                            match = TermQuery(field, encodedString, NonAnalyzer);
+                            continue;
+                        }
+
+                        match = @operator is SearchOperator.Or
+                            ? Or(match, TermQuery(field, encodedString))
+                            : And(match, TermQuery(field, encodedString));
+                        break;
+                    case SearchMatchOptions.StartsWith:
+                        if (match is null)
+                        {
+                            match = StartWithQuery(field, encodedString);
+                            continue;
+                        }
+
+                        match = @operator is SearchOperator.Or
+                            ? Or(match, StartWithQuery(field, encodedString))
+                            : And(match, StartWithQuery(field, encodedString));
+                        break;
+                    case SearchMatchOptions.EndsWith:
+                        if (match is null)
+                        {
+                            match = EndsWithQuery(field, encodedString);
+                            continue;
+                        }
+
+                        match = @operator is SearchOperator.Or
+                            ? Or(match, EndsWithQuery(field, encodedString))
+                            : And(match, EndsWithQuery(field, encodedString));
+                        break;
+                    case SearchMatchOptions.Contains:
+                        if (match is null)
+                        {
+                            match = ContainsQuery(field, encodedString);
+                            continue;
+                        }
+
+                        match = @operator is SearchOperator.Or
+                            ? Or(match, ContainsQuery(field, encodedString))
+                            : And(match, ContainsQuery(field, encodedString));
+                        break;
+                    default:
+                        throw new InvalidExpressionException("Unknown flag inside Search match.");
+                }
+            }
+            
+            QueryContext.MatchesPool.Return(buffer);
+            return match;
+        }
+
+        public IQueryMatch SearchQuery<TScoreFunction>(string field, string searchTerm, TScoreFunction scoreFunction, SearchOperator @operator, int analyzerId)
```

Comment:
why is this duplicated from the previous one?

Reply:
@redknightlois introduced two versions of each function with/without boosting. I'm keeping the convention for now.